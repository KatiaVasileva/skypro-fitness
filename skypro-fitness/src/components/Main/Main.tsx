import { cards } from "../../lib/data";
import Card from "../Card/Card";

function Main() {
  
  return (
    <div className="container">
      <div className="flex flex-row mt-2 md:mt-16 h-30 gap-4">
        <div className="h-30 basis-full md:basis-3/4">
          <h1 className="">
            Начните заниматься спортом и улучшите качество жизни
          </h1>
        </div>
        <div className="hidden md:block h-25 basis-0 md:basis-1/4 relative">
          <div className="bg-green text-3xl p-5 md:w-72 h-25 rounded mb-12 text-black">
            Измени своё тело за полгода!
          </div>
          <img
            className="absolute top-[100px] left-[120px] z-10"
            src="/img/triangle.png"
          />
        </div>
      </div>

      <div className="flex flex-row flex-wrap gap-4 md:gap-9 mb-8 mt-9 md:mt-8">
        {cards.map((course) => (
          <Card course={course} key={course.id} />
        ))}
      </div>

      <div className="flex justify-end md:justify-center ">
        <button className="btn-primary w-32">
          <a href="#top">Наверх ↑</a>
        </button>
      </div>
    </div>
  );
}

export default Main;

