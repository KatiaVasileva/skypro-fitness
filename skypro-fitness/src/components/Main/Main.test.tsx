/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Main from "./Main";
import UserProvider from "../../context/UserContext";
import CoursesProvider from "../../context/CoursesContext";

describe("Main component", () => {
  it("renders correctly", () => {
    const { container } = render(
    <UserProvider>
      <CoursesProvider>
        <Main />
      </CoursesProvider>
    </UserProvider>);
    expect(container).toMatchSnapshot();
  });
}); 