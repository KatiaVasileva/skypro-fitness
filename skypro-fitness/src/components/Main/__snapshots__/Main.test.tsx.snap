// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="flex flex-row mt-2 md:mt-16 h-30 gap-4"
    >
      <div
        class="h-30 basis-full md:basis-3/4"
      >
        <h1
          class="text-3xl md:text-4xl lg:text-6xl font-medium text-black"
        >
          Начните заниматься спортом и улучшите качество жизни
        </h1>
      </div>
      <div
        class="hidden md:block h-25 basis-0 md:basis-1/4 relative"
      >
        <div
          class="bg-green text-3xl p-5 md:w-72 h-25 rounded mb-12 text-black"
        >
          Измени своё тело за полгода!
        </div>
        <img
          class="absolute top-[100px] left-[120px] z-10"
          src="/img/triangle.png"
        />
      </div>
    </div>
    <div
      class="flex flex-row flex-wrap gap-4 md:gap-9 mb-8 mt-9 md:mt-8"
    />
    <div
      class="flex justify-end md:justify-center "
    >
      <button
        class="btn-primary w-32"
      >
        <a
          href="#top"
        >
          Наверх ↑
        </a>
      </button>
    </div>
  </div>
</div>
`;
